code/parachain/frame/staking-rewards/src/test/prelude.rs

```diff
@@ -100,47 +107,438 @@ where
+	should_be_early_unstake: bool,
+) where
 	Runtime: crate::Config<Event = RuntimeEvent> + frame_system::Config<Event = RuntimeEvent>,
 	RuntimeEvent: Parameter + Member + core::fmt::Debug + Clone,
 	RuntimeEvent: TryInto<crate::Event<Runtime>>,
 	<RuntimeEvent as TryInto<crate::Event<Runtime>>>::Error: core::fmt::Debug,
 	<Runtime as frame_system::Config>::Origin: OriginTrait<AccountId = AccountIdOf<Runtime>>,
 {
-	let existing_stake_before_split =
+	let position_before_unstake =
 		Stakes::<Runtime>::get(fnft_collection_id, fnft_instance_id).unwrap();
 
-	let (
-		(
-			event_existing_fnft_collection_id,
-			event_existing_fnft_instance_id,
-			existing_position_staked_amount,
-		),
-		(event_new_fnft_collection_id, event_new_fnft_instance_id, new_position_staked_amount),
-	) = assert_extrinsic_event_with::<Runtime, RuntimeEvent, crate::Event<Runtime>, _, _, _>(
-		crate::Pallet::<Runtime>::split(
-			OriginFor::<Runtime>::signed(staker),
+	let slashed_amount_of = |amount: Runtime::Balance| {
+		position_before_unstake.lock.unlock_penalty.left_from_one().mul_floor(amount)
+	};
+
+	let owner_staked_asset_balance_before_unstake =
+		Runtime::Assets::balance(position_before_unstake.reward_pool_id, &owner);
+
+	let rewards_pool = Pallet::<Runtime>::pools(position_before_unstake.reward_pool_id)
+		.expect("rewards_pool expected");
+
+	let pool_account_rewards_balances_before_unstake = rewards_pool
+		.rewards
+		.clone()
+		.into_iter()
+		.map(|(reward_asset_id, _)| {
+			(
+				reward_asset_id,
+				Runtime::Assets::balance(
+					reward_asset_id,
+					&Pallet::<Runtime>::pool_account_id(&position_before_unstake.reward_pool_id),
+				),
+			)
+		})
+		.collect::<BTreeMap<_, _>>();
+
+	let owner_rewards_balances_before_unstake = rewards_pool
+		.rewards
+		.clone()
+		.into_iter()
+		.map(|(reward_asset_id, _)| {
+			(reward_asset_id, Runtime::Assets::balance(reward_asset_id, &owner))
+		})
+		.collect::<BTreeMap<_, _>>();
+
+	let treasury_rewards_balances_before_unstake = rewards_pool
+		.rewards
+		.clone()
+		.into_iter()
+		.map(|(reward_asset_id, _)| {
+			(
+				reward_asset_id,
+				Runtime::Assets::balance(reward_asset_id, &Runtime::TreasuryAccount::get()),
+			)
+		})
+		.collect::<BTreeMap<_, _>>();
+
+	assert_extrinsic_event_with::<Runtime, RuntimeEvent, crate::Event<Runtime>, _, _, _>(
+		Pallet::<Runtime>::unstake(
+			OriginFor::<Runtime>::signed(owner.clone()),
 			fnft_collection_id,
 			fnft_instance_id,
-			ratio,
 		),
 		|event| match event {
-			crate::Event::SplitPosition { positions } =>
-				if let [existing, new] = positions[..] {
-					Some((existing, new))
+			crate::Event::Unstaked {
+				owner: event_owner,
+				fnft_collection_id: event_fnft_collection_id,
+				fnft_instance_id: event_fnft_instance_id,
+				slash,
+			} => {
+				if should_be_early_unstake {
+					assert!(slash.is_some(), "unstake was expected to be slashed but it was not");
+					assert_eq!(
+						slash.unwrap(),
+						position_before_unstake
```

Comment:
Though these tests are good to verify the current logic, they are in effect "too implementation specific", they provide no way to read(for a human) if a certain value has been calculated according the functional spec of a reward rate/slash etc. 
The tests generally should take in pool parameters and the output expected by a human, rather than automatically calculating output values.

Reply:
Something we can look at is also substrate pallet unit tests https://github.com/paritytech/substrate/blob/a47f200eebeb88a5bde6f1ed2be9728b82536dde/frame/staking/src/tests.rs , They tend to be very readable in the inputs and outputs. We need to get to a balance point of readability, automation and easy setup.

Reply:
Agreed - these tests & the helper functions will be improved in a followup PR